actors/miner/tests/extend_sector_expiration_test.rs

```diff
@@ -352,3 +365,413 @@ fn supports_extensions_off_deadline_boundary() {
+            }],
+        }],
+    };
+
+    let ae = h
+        .extend_sectors2(&mut rt, params, claims)
+        .expect_err("extend2 should error when claim id not found");
+    assert_eq!(ExitCode::USR_ILLEGAL_ARGUMENT, ae.exit_code());
+    // assert sector expiration is set to the new value
+    check_for_expiration(
+        &mut h,
+        &mut rt,
+        old_sector.expiration,
+        old_sector.sector_number,
+        deadline_index,
+        partition_index,
+    );
+}
+
+#[test]
+fn update_expiration_bad_claim() {
+    let (mut h, mut rt) = setup();
+    // add in verified deal
+    let verified_deals = vec![
+        TestVerifiedDeal { space: PaddedPieceSize(h.sector_size as u64 / 2) },
+        TestVerifiedDeal { space: PaddedPieceSize(h.sector_size as u64 / 2) },
+    ];
+    let old_sector = commit_sector_verified_deals(&verified_deals, &mut h, &mut rt);
+    h.advance_and_submit_posts(&mut rt, &vec![old_sector.clone()]);
+
+    let state: State = rt.get_state();
+
+    let (deadline_index, partition_index) =
+        state.find_sector(rt.policy(), rt.store(), old_sector.sector_number).unwrap();
+
+    let extension = 42 * rt.policy().wpost_proving_period;
+    let new_expiration = old_sector.expiration + extension;
+
+    let claim_ids = vec![400, 500];
+    let client = Address::new_id(3000).id().unwrap();
+
+    let fake_claim0 = make_fake_claim(
+        claim_ids[0],
+        &old_sector,
+        client,
+        h.receiver.id().unwrap(),
+        new_expiration,
+        &verified_deals[0],
+        rt.policy.minimum_verified_allocation_term,
+    );
+    // claim1 has a max term < new expiration so extension to new expiration should fail
+    let fake_claim1 = make_fake_claim(
+        claim_ids[1],
+        &old_sector,
+        client,
+        h.receiver.id().unwrap(),
+        new_expiration - 1,
+        &verified_deals[1],
+        rt.policy.minimum_verified_allocation_term,
+    );
+
+    let mut claims = HashMap::new();
+    claims.insert(claim_ids[0], Ok(fake_claim0));
+    claims.insert(claim_ids[1], Ok(fake_claim1));
+
+    let params = ExtendSectorExpiration2Params {
+        extensions: vec![ExpirationExtension2 {
+            deadline: deadline_index,
+            partition: partition_index,
+            sectors: make_bitfield(&[]),
+            new_expiration,
+            sectors_with_claims: vec![SectorClaim {
+                sector_number: old_sector.sector_number,
+                maintain_claims: claim_ids,
+            }],
+        }],
+    };
+
+    let ae = h
+        .extend_sectors2(&mut rt, params, claims)
+        .expect_err("extend2 should error when claim term max is forbidden");
+    assert_eq!(ExitCode::USR_FORBIDDEN, ae.exit_code());
+    // assert sector expiration is set to the new value
+    check_for_expiration(
+        &mut h,
+        &mut rt,
+        old_sector.expiration,
+        old_sector.sector_number,
+        deadline_index,
+        partition_index,
+    );
+}
```

Comment:
Nit: consider if you can re-use more code and set-up by using a common set-up for these tests, saving the state CID, then resetting to that state to check handling of various claims and parameters.